fe-library-app/src/components/AppRouter/pages/HomePage.tsx

```diff
@@ -1,13 +1,74 @@
+import { useEffect, useState } from 'react'
+
+import InfiniteScroll from 'react-infinite-scroll-component'
+
 import styles from './HomePage.module.css'
 import SearchBar from '../../SearchBar/SearchBar'
+import BookList from '../../BookList/BookList'
+import { GetBookResponse, getBooks, GetBooksRequest } from '../../../services/BookService'
 
 const HomePage = () => {
+  const [ bookList, setBookList ] = useState<GetBookResponse[]>([])
+  const [ search, setSearch ] = useState('')
+  const [ pageNumber, setPageNumber ] = useState(1)
+  const [ isResponseEmpty, setIsResponseEmpty ] = useState(true)
+
+  const getBooksPageAppend = (request: GetBooksRequest) => {
+    getBooks(request)
+      .then((response) => {
+        setBookList(prevList => [ ...prevList, ...response.data ])
+        setIsResponseEmpty(response.data.length === 0)
+      })
+      .catch(() => alert('Something went wrong!'))
+  }
+
+  const getBooksPage = (request: GetBooksRequest) => {
+    getBooks(request)
+      .then((response) => {
+        setBookList(response.data)
+        setIsResponseEmpty(response.data.length === 0)
+      })
+      .catch(() => alert('Something went wrong!'))
+  }
+
+  useEffect(() => {
+    getBooksPageAppend({ PageNumber: pageNumber, PageLength: 12, Where: [ { Field: 'Title', Value: search, Operation: 2 } ] })
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [ pageNumber ])
+
+  useEffect(() => {
+    setBookList([])
+    getBooksPage({ PageNumber: 1, PageLength: 12, Where: [ { Field: 'Title', Value: search, Operation: 2 } ] })
+    setPageNumber(2)
+  }, [ search ] )
+
+  const handleNextPage = () => {
+    setPageNumber(prevPage => prevPage + 1)
+  }
+
+  const searchChangeHandler = (newInput: string): void => {
+    setSearch(newInput)
+  }
+
   return (
     <div className={styles.home}>
       <div className={styles.screen_search}>
-        <SearchBar />
+        <SearchBar searchChange={searchChangeHandler}/>
+      </div>
+      <h1 className={styles.home_content}>Books: </h1>
+      <div className={styles.inf_wrap}>
+        <InfiniteScroll
+          dataLength={bookList.length}
+          next={handleNextPage}
+          hasMore={!isResponseEmpty}
```

Comment:
How is has more false only if it's an empty response for previous request? That means we always have an extra request. Has more should be calculated based on total elements, current page and num of elements per page

Reply:
For knowing total elements, I need to send an extra request for get all elements and then use it's length, because paged elements doesn't give us total count of it. Having that in mind, both solutions need same amount of requests.